### PR TITLE
Improve documetation of nml XML

### PR DESCRIPTION
**Pull request by jgfouca**
This was driven by comments that were stranded in the last PR.

### CODE REVIEW

File: components/scream/cime_config/buildnml
Comment (bartgol) [inline components/scream/cime_config/buildnml:9]: Do we really want to let them modify it directly? I mean, we can't prevent this from happening, but we should *highly* discourage it. If the user accidentally modifies the structure of the file, we might generate something completely off, with an invalid xml format being our best-case scenario.

**Comment by PeterCaldwell** [comment]
Looks great! I'm not approving now just to give others a chance to comment before auto-merge.

**Comment by jeff-cohere** [comment]
> Looks great! I'm not approving now just to give others a chance to comment before auto-merge.

At least we learn from each other's goofs.

File: components/scream/cime_config/namelist_defaults_scream.xml
Comment (jeff-cohere) [inline components/scream/cime_config/namelist_defaults_scream.xml:34]: evaulate -> evaluate

Comment (bartgol) [inline components/scream/cime_config/namelist_defaults_scream.xml]: I think the comment might be misleading for some one. It's not the last selector within a single element that takes precedence, but the last element among those (with the same name) that have all selectors that eval to true. I think a mini example might help. Something tiny, like
```
If bar=3 and baz=4, then
   <foo bar="3">my_foo</foo>
   <foo baz="4">your_foo</foo>
would generate an xml with content
   <foo>your_foo</foo> 
```
And maybe we should add a warning, cause this can lead to defaults being inadvertently shadowed, depending on the order in scream_namelists_defaults.xml

**Comment by jeff-cohere** [comment]
Looks good to me, too. I found one typo.

File: components/scream/cime_config/namelist_defaults_scream.xml
Comment (bartgol) [reply components/scream/cime_config/namelist_defaults_scream.xml]: Actually, we can probably remove this sentence, since you go over the rules later, and rule 1 clearly states this fact already.

Comment (jgfouca) [reply components/scream/cime_config/namelist_defaults_scream.xml]: Done.

**Review by bartgol — COMMENTED** [review]
Looks good. Got a couple of comments, but nothing to hold this back.

**Comment by jgfouca** [comment]
What the heck? How did AT fail with AT: Integrate without testing?

**Comment by bartgol** [comment]
> What the heck? How did AT fail with AT: Integrate without testing?

And all builds passed!

**Comment by jgfouca** [comment]
@jeff-cohere , I just realized you had a lot of comments in the older PR that I did not notice since they were under "hidden items". I'm going through them now.

**Comment by bartgol** [comment]
> What the heck? How did AT fail with AT: Integrate without testing?

Ah, you pushed a commit to the PR right after the builds finished, but before the AT was done with the PR, so it marked it as failed, to prevent merging.

**Comment by jgfouca** [comment]
Ah, unlucky race condition.

File: components/scream/cime_config/buildnml
Comment (jgfouca) [reply components/scream/cime_config/buildnml:9]: Remember, with the changes introduced yesterday to the prototype, the metadata of parameters, computed from the repo XML file, must be obeyed in the $case file. So it's not completely left up to the user to not make a mistake.

File: components/scream/cime_config/namelist_defaults_scream.xml
Comment (jgfouca) [reply components/scream/cime_config/namelist_defaults_scream.xml]: I removed the sentence; it wasn't even using the right terminology.

**Comment by jgfouca** [comment]
OK, as far as I can tell, this is all ready. Can someone approve?

**Review by jeff-cohere — APPROVED** [review]

**Pull request merged**
